#! /usr/bin/env python

PACKAGE='htp_auto'

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
#       Name              Type      Level            Description           Default  Min  Max
gen.add("yaw_offset",   double_t,    0,    "GPS offset due to mounting (deg)", 90, -180, 180)
gen.add("max_pdop",     double_t,    0,    "Maximum Position Dilution",        5,  0,  100)
gen.add("min_sats",     int_t,       0,    "Minimum Number of Satellites.",    8,  0,  20)

gen.add("yaw_min_qual_ind", int_t,   0,    "Minimum Fix Quality Indicator for Yaw.",   3,  0,  4)
gen.add("yaw_max_qual_ind", int_t,   0,    "Maximum Fix Quality Indicator for Yaw.",   3,  0,  4)

gen.add("pos_qual_ind_1", int_t,   0,    "Acceptable Fix Quality Indicator 1.",   3,  0,  13)
gen.add("pos_qual_ind_2", int_t,   0,    "Acceptable Fix Quality Indicator 2.",   3,  0,  13)
gen.add("pos_qual_ind_3", int_t,   0,    "Acceptable Fix Quality Indicator 3.",   3,  0,  13)

gen.add("dist_to_center_of_rotation", double_t, 0, "In meters.",  .2,  -2,  2)

exit(gen.generate(PACKAGE, "gps_converter", "GPSConverterParams"))
